scripts/install-linux.sh

```
@@ -296,12 +300,16 @@ if command -v $INSTALL_DIR/$BINARY_NAME &> /dev/null; then
         echo "$PROVIDER_GITLAB" >> /etc/opk/providers
     fi
 
-    AUTH_KEY_CMD="AuthorizedKeysCommand /usr/local/bin/opkssh verify %u %k %t"
+    AUTH_KEY_CMD="AuthorizedKeysCommand ${INSTALL_DIR}/opkssh verify %u %k %t"
     AUTH_KEY_USER="AuthorizedKeysCommandUser ${AUTH_CMD_USER}"
 
     # Add the directives in the correct configuration, taking priorities into account
-    if ! grep -q '^Include /etc/ssh/sshd_config\.d/\*\.conf' /etc/ssh/sshd_config \
-    || ! grep -q '^AuthorizedKeysCommand\|^AuthorizedKeysCommandUser' /etc/ssh/sshd_config.d/*.conf ; then
+    if [[ -f /etc/ssh/sshd_config ]] && \
```

If I'm reading this correct, this code has the behavior that If `/etc/ssh/sshd_config` can not be found it jumps to the else and tries to create a `/etc/ssh/sshd_config.d/*` file. We should cover the case that ``/etc/ssh/sshd_config` doesn't exist and ``/etc/ssh/sshd_config.d/` doesn't exist. What happens if sshd isn't installed?

We should probably fail early and inform the user they need to install sshd or that no config can be found in the `/etc/ssh/` directory

---

That makes sense, I could had that check as well.
I think a better way to check that is to use the package manager to check if that package is installed or not early in the script.
Something like (need to test it though):
```
check_openssh_installed() {
    case "$OS_TYPE" in
        redhat)
            if rpm -q openssh-server &>/dev/null; then
                return 0
            else
                echo "OpenSSH server is NOT installed."
                echo "To install it, run: sudo dnf install openssh-server"
                exit 1
            fi
            ;;
        debian)
            if dpkg -l | grep -q '^ii.*openssh-server'; then
                return 0
            else
                echo "OpenSSH server is NOT installed."
                echo "To install it, run: sudo apt install openssh-server"
                exit 1
            fi
            ;;
        arch)
            if pacman -Q openssh &>/dev/null; then
                return 0
            else
                echo "OpenSSH server is NOT installed."
                echo "To install it, run: sudo pacman -S openssh"
                exit 1
            fi
            ;;
        suse)
            if rpm -q openssh-server &>/dev/null; then
                return 0
            else
                echo "OpenSSH server is NOT installed."
                echo "To install it, run: sudo zypper install openssh"
                exit 1
            fi
            ;;
    esac
}
```

---

That makes sense. 

I just want to make sure if those config directories don't exist for some strange reason, the install fails as it would be very difficult to debug after the fact if it succeeds despite sshd config actually living somewhere else.

---

Fair enough,
I could add both checks and a give a  explanatory error message if the package is installed but both the config file directory are missing. That way the install script catches the weird scenario when the config file exists but the openSSH server isn't installed.

---

@EthanHeilman , I've updated the PR description and added the openssh-server package check and configuration check to the script.